From 901ff183158a7294fdcd7498581129940536c542 Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Tue, 14 Feb 2012 09:10:53 +0800
Subject: ENGR00174394 MX6Q_arm2/MX6Q_sabreauto: change ipu_id/disp_id for LDB
 config.

MX6Q_arm2/MX6Q_sabreauto:  change ipu_id/disp_id for LDB configuration.
For, LDB_SEP0 mode, the disp_id should be 0, and sec_disp_id should be
1 on MX6Q, since the LDB channel 0 should be connected to IPU DI0.

Signed-off-by: Wayne Zou <b36644@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c      |    8 ++++----
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |    4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 71161e8..a038038 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -1390,12 +1390,12 @@ static struct fsl_mxc_lcd_platform_data lcdif_data = {
 };
 
 static struct fsl_mxc_ldb_platform_data ldb_data = {
-	.ipu_id		= 0,
-	.disp_id	= 1,
+	.ipu_id		= 1,
+	.disp_id	= 0,
 	.ext_ref	= 1,
 	.mode		= LDB_SEP0,
-	.sec_ipu_id	= 1,
-	.sec_disp_id	= 0,
+	.sec_ipu_id	= 0,
+	.sec_disp_id	= 1,
 };
 
 static struct imx_ipuv3_platform_data ipu_data[] = {
diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index 8b582d5..525e0bd 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -1009,8 +1009,8 @@ static struct fsl_mxc_lcd_platform_data lcdif_data = {
 };
 
 static struct fsl_mxc_ldb_platform_data ldb_data = {
-	.ipu_id = 0,
-	.disp_id = 1,
+	.ipu_id = 1,
+	.disp_id = 0,
 	.ext_ref = 1,
 	.mode = LDB_SEP0,
 	.sec_ipu_id = 1,
-- 
1.7.9.5

